### PR TITLE
ARTEMIS-2076 Make Filter updateable

### PR DESCRIPTION
**Pull request by michaelandrepearce**

### CODE REVIEW

**Comment by clebertsuconic** [comment]
Is this updating the queue on the journal as well ?

**Comment by clebertsuconic** [comment]
It is not !!!   This will need to be a more complete feature.   If you restart the broker you get the old filter back afaik.

File: artemis-core-client/src/main/java/org/apache/activemq/artemis/api/core/management/ActiveMQServerControl.java [hunk 1]
```diff
@@ -710,6 +710,7 @@ String updateQueue(@Parameter(name = "name", desc = "Name of the queue") String
                       @Parameter(name = "exclusive", desc = "If the queue should route exclusively to one consumer") Boolean exclusive,
                       @Parameter(name = "consumersBeforeDispatch", desc = "Number of consumers needed before dispatch can start") Integer consumersBeforeDispatch,
                       @Parameter(name = "delayBeforeDispatch", desc = "Delay to wait before dispatching if number of consumers before dispatch is not met") Long delayBeforeDispatch,
+                      @Parameter(name = "filter", desc = "The filter to use on the queue") String filter,
```
[the remark this answers is not in the document]
Comment (michaelandrepearce) [reply artemis-core-client/src/main/java/org/apache/activemq/artemis/api/core/management/ActiveMQServerControl.java]: This change is in the same release e.g. new method (the delay before dispatch stuff is new)

**Comment by michaelandrepearce** [comment]
@clebertsuconic its updating as per other queue field update.... see updateQueue in postoffice.

**Comment by michaelandrepearce** [comment]
if (filter != null && !filter.equals(queue.getFilter())) {
            changed = true;
            queue.setFilter(filter);
         }
         if (logger.isDebugEnabled()) {
            if (user == null && queue.getUser() != null) {
               logger.debug("Ignoring updating Queue to a NULL user");
            }
         }
         if (user != null && !user.equals(queue.getUser())) {
            changed = true;
            queue.setUser(user);
         }

         if (changed) {
            final long txID = storageManager.generateID();
            try {
               storageManager.updateQueueBinding(txID, queueBinding);
               storageManager.commitBindings(txID);
            } catch (Throwable throwable) {
               storageManager.rollback(txID);
               logger.warn(throwable.getMessage(), throwable);
               throw throwable;
            }
         }

File: artemis-core-client/src/main/java/org/apache/activemq/artemis/api/core/management/ActiveMQServerControl.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jbertram) [reply artemis-core-client/src/main/java/org/apache/activemq/artemis/api/core/management/ActiveMQServerControl.java]: Got it.  Thanks for the clarification.

**Comment by jbertram** [comment]
IMO all these commits should be squashed.  I don't mind if you take authorship of the test I wrote for the other PR (assuming you don't).

**Comment by michaelandrepearce** [comment]
@jbertram yes, i was wondering about squashing and then worried about your commit, as you dont mind, ill squash.

**Comment by michaelandrepearce** [comment]
@jbertram all squashed now.

**Comment by clebertsuconic** [comment]
@michaelandrepearce as I talked on the vote thread: Can you tweak (or add a test) that play with a broker restart, and validate the queue is updated on the journal, and messages not lost please?

**Comment by michaelandrepearce** [comment]
@clebertsuconic the issue on vote thread is around route-type change from another JIRA and PR, this is separate feature, though we're avoiding the issue that introduced, by following the same update logic of other queue field updates.

I will add a test case though to ensure when running and config is reloaded the queue is not destroyed but updated instead e.g. ensure the queue instance is the same.

**Comment by jbertram** [comment]
LGTM

**Comment by michaelandrepearce** [comment]
Dont merge i need to extend the tests still as per cleberts request

**Comment by clebertsuconic** [comment]
@michaelandrepearce if you don't have time to do it don't worry I will do it.. I was asking more in a way like.. if you can... know what I mean?

**Comment by michaelandrepearce** [comment]
@clebertsuconic i have enhanced the tests to check binding ids are the same (aka not recreated) and also added check for message loss, this is added both to the redeploy and the restart test cases.

**Comment by clebertsuconic** [comment]
I was able to fix the conflicts myself after the previous merges.. don't worry!

**Comment by michaelandrepearce** [comment]
@clebertsuconic i can rebase if you need to merge this? I have one locally already, let me know if you wish me to push up, or if you're merging as is (i assume from your response the latter)

**Comment by clebertsuconic** [comment]
Don’t worry. It was easy to rebase. Simple one.   


Will merge it soon.

**Pull request closed without merging**